src/field.ml

```diff
@@ -1,12 +1,18 @@
 include Field_intf
 
-type kind = _kind [@@deriving repr]
+type nonrec kind = kind [@@deriving repr]
 
 module MakeInt (Size : SIZE) = struct
   type t = int [@@deriving repr]
 
   let size = Size.size
 
+  (* Your [Int(32,64).(to,of)_int] uses are unsafe. You should do something about it.
+
+     [get_int32_be b off |> Int32.to_int] would be wrong on large enough ints on 32bit systems
+
+     [Int32.of_int |> set_int32_be b off] would be wrong on large enough ints
```

Comment:
Does `Int32.of_int` not fail with a large enough int ? if so I agree this is unsafe, although if we attempt to `set` an int that does not fit on 4 bytes, while `size` is 4, then something has already gone wrong.

Reply:
```
> echo 'Int32.of_int 9999999999999;;' | ocaml
# - : int32 = 1316134911l
```